### ISSUE TITLE
Quit button closes immediately

### ISSUE THREAD

**Issue by Huite** [rchartra/NcCut#38]
Related to: https://github.com/openjournals/joss-reviews/issues/7185

If you click the Quit Button it closes immediately. In most programs, quitting via the in-application menu is two or three steps:

* Click File Menu (or something similar)
* Click Close
* If there's unsaved work, it will now still show a confirmation dialog)

In NcCut, the Quit button is right there on screen, and it closes the app immediately. This means there's no room for a misclick.
I'd suggest opening a confirmation dialog giving the user the option to cancel in case of a misclick.

**Comment by platipodium** [comment]
I think I disagree with @Huite on the solution, but not on the problem analysis.   The sudden/unexpected closing of the app is very well in line with ncview's behavior :=(. On the other hand, it is a quicklook tool, and as such should be quick to dismiss, as well.  I would feel obstructed by a "do you really want to close?" dialog.

If there is no data loss on quick-close (e.g. by auto-saving the view) I'd have no problem with the one-click close functionality.

**Comment by rchartra** [comment]
One possibility is to show the confirmation dialog, but have an option to select "Don't show this again" so the user can choose to one-click quit on future occasions.

**Comment by Huite** [comment]
I'm often somewhat annoyed by the "Are you sure?" prompts too... and this is a quicklook tool indeed.

Some thoughts:

* I think I opened this issue specifically because I misclicked, which happens easily because the Quit button is near the other buttons.
* The application can be also closed directly via either the taskbar, or via the `X` in the window corner. These are much less likely to be hit accidentally. I use several application which don't even have an in-GUI quit button.
* A "File" menu with a Quit option is already two clicks, and thereby guards against misclicks, but that's not a good fit for NcCut.
* The "Don't show this again" sounds reasonable, but you'll have to start storing user settings and for most applications that do this, I wouldn't know how to re-enable showing the dialog.
